## IMDb- Web Scraping

### Importing important libraries

In [1]:
from bs4 import BeautifulSoup
import requests,openpyxl

### Creating an excel file to store the scraped data

In [2]:
excel=openpyxl.Workbook()
print(excel.sheetnames)
sheet=excel.active
sheet.title= "Top Rated Movies"
sheet.append(["Movie Rank","Movie Name","Year of release"," IMDB Rating"])

['Sheet']


### The url of the IMDB website we want to scrape

In [6]:
    source =requests.get('https://www.imdb.com/chart/top/')
    source.raise_for_status()
    
    soup=BeautifulSoup(source.text,'html.parser')
    
    movies=soup.find('tbody',class_="lister-list").find_all('tr')
    print(movies)

[<tr>
<td class="posterColumn">
<span data-value="1" name="rk"></span>
<span data-value="9.236244920191663" name="ir"></span>
<span data-value="7.791552E11" name="us"></span>
<span data-value="2753767" name="nv"></span>
<span data-value="-1.7637550798083375" name="ur"></span>
<a href="/title/tt0111161/"> <img alt="The Shawshank Redemption" height="67" src="https://m.media-amazon.com/images/M/MV5BNDE3ODcxYzMtY2YzZC00NmNlLWJiNDMtZDViZWM2MzIxZDYwXkEyXkFqcGdeQXVyNjAwNDUxODI@._V1_UX45_CR0,0,45,67_AL_.jpg" width="45"/>
</a> </td>
<td class="titleColumn">
      1.
      <a href="/title/tt0111161/" title="Frank Darabont (dir.), Tim Robbins, Morgan Freeman">The Shawshank Redemption</a>
<span class="secondaryInfo">(1994)</span>
</td>
<td class="ratingColumn imdbRating">
<strong title="9.2 based on 2,753,767 user ratings">9.2</strong>
</td>
<td class="ratingColumn">
<div class="seen-widget seen-widget-tt0111161 pending" data-titleid="tt0111161">
<div class="boundary">
<div class="popover">
<span 

 As we can see the HTML code, the data can be scraped from this code.

In [7]:
print(len(movies))

250


There are total of 250 movies

### The data we will extract from IMDB website  is : Movie rank, Movie name, Year of release and IMDB Rating 

#### First we are scraping one element from the website and accordingly all the data will be extracted.

1.Right click on the data we want to extract and click on Inspect.
2.A window with HTML code will appear on the right side of the webpage.
3.Find the container in which the data we want to extract is stored.

#### Extracting first movie name

In [ ]:
name=movie.find('td',class_="titleColumn").a.text

#### Similarly, we will extract rest of the information for the first movie.

In [ ]:
# Extracting rank of the movie
rank=movie.find('td',class_="titleColumn").get_text(strip=True).split(".")[0]

#### Extracting year of release

In [ ]:
year=movie.find('td',class_="titleColumn").span.text.strip('()')

#### The last thing we need to extract is imdb ratinngs.

In [ ]:
rating=movie.find('td',class_="ratingColumn imdbRating").strong.text

#### We have total of 250 movies. So, instead of specifying URL each time, we will write a loop to extract the complete 250 rows and store it in the sheet

In [ ]:
for movie in movies:
        name=movie.find('td',class_="titleColumn").a.text
        rank=movie.find('td',class_="titleColumn").get_text(strip=True).split(".")[0]
        year=movie.find('td',class_="titleColumn").span.text.strip('()')
        rating=movie.find('td',class_="ratingColumn imdbRating").strong.text
        print(rank,name,year,rating)
        sheet.append([rank,name,year,rating])

### Save the excel file 

In [5]:
excel.save('IMDB Movie Rating.xlsx')